Cmake improvements

hakostra opened the pull request:
This PR fix CMake such that the Blosc library is fetched and compiled automatically (as intended, just a typo fixed). It also adds the dynamic HDF5 plugin (libH5Zblosc) as a target. This is enabled by default, but can optionally be disabled. This PR close issue #3.

Commit:
Fixed dependency typo/issue

Fixed typo error in dependency: hdf5-blosc now pulls Blosc from Git and builds it as intended.

Commit:
Dynamic HDF5 plugin compiled by CMake

CMake now compile the dynamically loadable plugin alongside the regular filter. The installation (i.e. moving the libH5Zblosc.so to the right location) is left to the user currently.

This fix closes issue #3.

FrancescAlted commented:
Uh, it would be nice if the `blosc_plugin_shared` is installed too, right?

hakostra commented:
Yes, it would be. But where to install it?
1. Hard-coded to the HDF5 plugin path "/usr/local/hdf5/lib/plugin"? This will create problems for those without root-access to their computers/sorkstations/clusters etc.
2. In the same location as the blosc_filter? HDF5 will not pick it up there.
3. To the environment-variable $HDF5_PLUGIN_PATH? This will require that variable to be set. My impression is that it is generally not set on the systems I regularly use.
4. Allowing the user to specify a custom location for the plugin? Will complicate the install process.

Opinions?

FrancescAlted commented:
I think copying it to /usr/local/hdf5/lib/plugin by default would be the correct way to go.  People should know that installing normally requires privileges, so I don't think this will come with a surprise.  Personally I'd define a new cmake symbol (say, HDF5_PLUGIN_PATH) and set it to /usr/local/hdf5/lib/plugin by default.  People will always be able to override it with `cmake -DHDF5_PLUGIN_PATH=their_place` if they want to.

Commit:
Plugin now installed by CMake

The installation location can be changed in cmake with '-DPLUGIN_INSTALL_PATH=/your/plugin/location' if you do not want to use the standard '/usr/local/hdf5/lib/plugin' location.

hakostra commented:
Done. I named the variable PLUGIN_INSTALL_PATH instead of HDF5_PLUGIN_PATH, to not confuse it with the HDF5_PLUGIN_PATH environment variable (which can be set to multiple paths, so it cannot be used directly). Defaults to "/usr/local/hdf5/lib/plugin".

FrancescAlted commented:
LGTM.  Thanks!

stevengj commented:
👍
